File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.js

function solution(common) {
    let dis;
    if (common[1] - common[0] === common[2] - common[1]) {
        return common[common.length - 1] + common[1] - common[0];
    }
    if (common[1] / common[0] === common[2] / common[1]) {
        return common[common.length - 1] * common[1] / common[0];
    }
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.java
class Solution {
    public int solution(int[] common) {
        int answer = 0;
        if(common[2] - common[1] == common[1] - common[0]){
            answer = common[1] - common[0] + common[common.length - 1];
        }else if(common[2] / common[1] == common[1] / common[0]){
            answer = common[1] / common[0] * common[common.length - 1];
        }
        return answer;
    }
}